### PR TITLE
DOCS-1069: Document API methods for `Session` and `SessionManager`

### PR DESCRIPTION
**Pull request by sguequierre**
* Add documentation for methods available and documented in RDK client docs

### CODE REVIEW

File: docs/program/apis/sessions.md [hunk 1]
```diff
@@ -126,6 +126,124 @@ disableSessions: true
...
+- `ctx` [(Context)](https://pkg.go.dev/context): A Context carries a deadline, a cancellation signal, and other values across API boundaries.
+
+For more information, see the [Go SDK Docs](https://pkg.go.dev/go.viam.com/rdk/session#ToContext).
+
+```go
+// Attach session "my session" to the given Context 
+session = session.ToContext(context.Background(), my_session)
+```
+
+## Manager API
```
Comment (sguequierre) [inline docs/program/apis/sessions.md:225]:
```suggestion
## Manager API
```

File: docs/program/apis/session-manager.md [hunk 2]
```diff
@@ -8,7 +8,8 @@ tags: ["client", "sdk", "viam-server", "networking", "apis", "robot api", "sessi
 ---
 
 When controlling a robot or fleet with Viam, you want a way to understand the presence of the clients that are communicating with and authenticated to each robot's `viam-server`.
-The period of time during which these clients are present is called a *session*.
+The period of time during which these clients are present is called a *session.*
+Sessions each have [their own API](/program/apis/session/), which is a client of `viam-server`.
```
Comment (sguequierre) [inline docs/program/apis/session-manager.md:12]:
```suggestion
Sessions each have [their own API](/program/apis/session/), which is a client of both `viam-server` and the [session management API](/program/apis/session-manager).
```

File: docs/program/apis/session-manager.md [hunk 3]
```diff
@@ -20,10 +23,10 @@ Session management allows for safer operation of robots that physically move.
 For example, imagine a wheeled rover gets a [`SetPower()`](/components/base/#setpower) command as the last input from a client before the connection to the robot is interrupted.
 Without session management, the API request from the client sets the flow of electricity to the motors of the robot and then does not time out, causing the robot to continue driving forever, colliding with objects and people.
 
-With default configuration, sessions are automatically managed for you with Viam's `SessionsClient`.
-If you want to manage sessions yourself, use Viam's sessions management API.
+With default configuration, sessions are automatically managed for you with Viam's session management API's default `SessionsClient`.
+If you want to manage sessions yourself, use Viam's session management API.
 
-### The `SessionsClient`
+### What is a `SessionsClient`
```
Comment (sguequierre) [inline docs/program/apis/session-manager.md:29]:
```suggestion
### What is a `SessionsClient`?
```

File: docs/program/apis/session-manager.md [hunk 4]
```diff
@@ -126,6 +129,113 @@ disableSessions: true
...
+{{< readfile "/static/include/program/apis/session-manager.md" >}}
+
+### Start
+
+Create a new session that expects at least one heartbeat within the configured window.
+
+**Parameters:**
+
+- `ctx` [(Context)](https://pkg.go.dev/context): A Context carries a deadline, a cancellation signal, and other values across API boundaries.
+- `heartbeatWindow` [(string)](https://pkg.go.dev/time#Duration): The heartbeat window you want this `SessionManager` to follow for this session. The window is the elapsed time between two instants as an [int64](https://pkg.go.dev/builtin#int64) nanosecond count. The representation limits the largest representable duration to approximately 290 years
```
Comment (sguequierre) [inline docs/program/apis/session-manager.md:149]:
```suggestion
- `heartbeatWindow` [(time.Duration)](https://pkg.go.dev/time#Duration): The heartbeat window you want this `SessionManager` to follow for this session. The window is the elapsed time between two instants as an [int64](https://pkg.go.dev/builtin#int64) nanosecond count. The representation limits the largest representable duration to approximately 290 years
```

File: docs/program/apis/session-manager.md [hunk 5]
```diff
@@ -126,6 +129,113 @@ disableSessions: true
...
+
+Get all sessions that are actively being held by this `SessionManager`.
+
+**Parameters:**
+
+- None
+
+**Returns:**
+
+- [([]*Session)](https://pkg.go.dev/go.viam.com/rdk@v0.8.0/session#Session): All active sessions associated with this `SessionManager`.
```
Comment (sguequierre) [inline docs/program/apis/session-manager.md:172]:
```suggestion
- [([]*Session)](https://pkg.go.dev/go.viam.com/rdk/session#Session): All active sessions associated with this `SessionManager`.
```

File: docs/program/apis/session-manager.md [hunk 6]
```diff
@@ -126,6 +129,113 @@ disableSessions: true
...
+**Parameters:**
+
+- `ctx` [(Context)](https://pkg.go.dev/context): A Context carries a deadline, a cancellation signal, and other values across API boundaries.
+- `id` [(uuid.UUID)](https://pkg.go.dev/github.com/google/uuid#UUID): The client's UUID associated with this session.
+- `ownerID` [(string)](https://pkg.go.dev/builtin#string): The client's ownerID associated with this session.
+If ownerID is in use but the session in question has a different owner, this is a security violation and Viam reports back that no session is found.
+
+**Returns:**
+
+- [(*Session)](https://pkg.go.dev/go.viam.com/rdk@v0.8.0/session#Session): A new manager for holding sessions.
```
Comment (sguequierre) [inline docs/program/apis/session-manager.md:194]:
```suggestion
- [(*Session)](https://pkg.go.dev/go.viam.com/rdk/session#Session): A new manager for holding sessions.
```

File: docs/program/apis/session-manager.md [hunk 7]
```diff
@@ -126,6 +129,113 @@ disableSessions: true
...
 {{% /tabs %}}
 
-This option allows you to have full control over sessions management.
+This option allows you to have full control over session management.
 After disabling the client, you must now manage each of your sessions manually with the session management API.
 You can do this with Viam's [client SDKs](https://pkg.go.dev/go.viam.com/rdk/session).
+
+## API
+
+The `SessionManager` API supports the following methods:
```
Comment (sguequierre) [inline docs/program/apis/session-manager.md:138]:
```suggestion
The `SessionManager` API built-in to the [Robot API](/program/apis/robot/) supports the following methods:
```

File: docs/program/apis/session-manager.md [hunk 8]
```diff
@@ -126,6 +129,113 @@ disableSessions: true
...
 {{% /tabs %}}
 
-This option allows you to have full control over sessions management.
+This option allows you to have full control over session management.
 After disabling the client, you must now manage each of your sessions manually with the session management API.
 You can do this with Viam's [client SDKs](https://pkg.go.dev/go.viam.com/rdk/session).
+
+## API
+
+The `SessionManager` API built-in to the [Robot API](/program/apis/robot/) supports the following methods:
```
Comment (sguequierre) [inline docs/program/apis/session-manager.md:138]: I think this needs documentation on `NewSessionManager()` in robot API to be functional-- sme review may help clear up some confusion on my end

File: docs/program/apis/session.md [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+description: "Safety monitor resources and attach contexts to sessions with the Session API."
+tags: ["client", "sdk", "viam-server", "networking", "apis", "robot api", "session", "sessions", "session management"]
+---
+
+When controlling a robot or fleet with Viam, you want a way to understand the presence of the clients that are communicating with and authenticated to each robot's `viam-server`.
+The period of time during which these clients are present is called a *session*.
+
+## API
+
+The `Session` API supports the following methods:
```
Comment (sguequierre) [inline docs/program/apis/session.md:15]: Question for @dgottlieb or any SME who has a better idea of what's going on in the RDK than I do: is this correct? Or should this all be replaced with https://pkg.go.dev/go.viam.com/rdk@v0.8.0/session#Session as these are just the methods the `session` package offers and the interface for the type itself has different methods (Active, Heartbeat, etc.). Should both be on here? Apologies for the confusion

File: docs/program/apis/session-manager.md (hunk 8 above)
Comment (sguequierre) [reply docs/program/apis/session-manager.md:138]: Added, then removed,`NewSessionManager()` instantiation of `mySessionManager`-- unsure whether that should be in there or not. How would a user most simply request to the default session manager API with what we have set up in the RDK? @dgottlieb

**Comment by sguequierre** [comment]
Closing out and leaving these methods undocumented as they are not meant to be user-facing re discussion with @dgottlieb @npentrel

File: docs/program/apis/session.md (hunk 9 above)
Comment (dgottlieb) [reply docs/program/apis/session.md:15]: Similarly, for posterity, these are internal data types that are not client facing.

**Review by dgottlieb — COMMENTED** [review]
FWIW, the following is the (internal) client* session/heartbeat code that gets run when connecting with sessions enabled:
* [Start/create a session](https://github.com/viamrobotics/rdk/blob/b39c81bab8768a06be9780d577fdc0e27be53ed5/robot/client/client_session.go#L122-L127)
* [Sending a heartbeat](https://github.com/viamrobotics/rdk/blob/b39c81bab8768a06be9780d577fdc0e27be53ed5/robot/client/client_session.go#L66-L69)
* [Adding session data](https://github.com/viamrobotics/rdk/blob/b39c81bab8768a06be9780d577fdc0e27be53ed5/robot/client/client_session.go#L89) to other RPC commands (e.g: `WheeledBase.MoveStraight`)

**Pull request closed without merging**